dali/ft/daftformat.ipp

```diff
@@ -473,27 +474,32 @@ public:
         }
         return true;
     }
+    bool checkFoundRowStart()
+    {
+        return (rowStart!=(offset_t)-1);
+    }
     offset_t getHeaderLength()
     {
         if (!headerLength)
         {
-            while (!rowStart && reader->next());
-            if (!rowStart)
+            while (!checkFoundRowStart() && reader->next());
+            if (!checkFoundRowStart())
                 throw MakeStringException(DFTERR_CannotFindFirstJsonRecord, "Could not find first json record (check path)");
             else
-                headerLength = rowStart-1;
+                headerLength = rowStart;
         }
         return headerLength;
     }
     offset_t getFooterLength()
     {
-        if (!footerLength)
+        if (footerLength==(offset_t)-1)
         {
+            footerLength = 0;
             while (reader->next());
             if (rowEnd)
             {
                 footerLength = isRootless() ? 0 : 1; //account for parser using ] as offset unless rootless
-                footerLength = footerLength + size - rowEnd;
+                footerLength = footerLength + size - rowEnd - 1;
```

Comment:
No, I made the parser more consistent with xml so that the virtual filepositions in my index were right..  that had a side effect of changine the way the offsets had to be handled in spraying, but I think its more correct.